src/nvim/api/private/dispatch.c

```diff
@@ -0,0 +1,47 @@
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <assert.h>
+#include <msgpack.h>
+
+#include "nvim/map.h"
+#include "nvim/log.h"
+#include "nvim/vim.h"
+#include "nvim/msgpack_rpc/helpers.h"
+#include "nvim/api/private/dispatch.h"
+#include "nvim/api/private/helpers.h"
+#include "nvim/api/private/defs.h"
+
+#include "nvim/api/buffer.h"
+#include "nvim/api/tabpage.h"
+#include "nvim/api/ui.h"
+#include "nvim/api/vim.h"
+#include "nvim/api/window.h"
+
+static Map(String, MsgpackRpcRequestHandler) *methods = NULL;
+
+static void msgpack_rpc_add_method_handler(String method, MsgpackRpcRequestHandler handler)
+{
+  map_put(String, MsgpackRpcRequestHandler)(methods, method, handler);
+}
+
+MsgpackRpcRequestHandler msgpack_rpc_get_handler_for(const char *name,
+                                                     size_t name_len)
+{
+  String m = {
+    .data=(char *)name,
+    .size=name_len, //TODO(bfredl): was the max_fname_len check important?
```

Comment:
It was an attempt at an optimization (c.f., the comment above the declaration of `max_fname_len`).  If it were to be kept, I'd suggest that instead of setting size to `MIN(name_len, max_fname_len)`, immediately set `rv.fn = msgpack_rpc_handle_missing_method;` and return.